### DISCUSSION TITLE
Understanding the nginx config to serve from a subdirectory

### DISCUSSION THREAD

**Discussion by jeanas** [LycheeOrg/Lychee#2711] [General]
[This FAQ](https://lycheeorg.github.io/docs/faq_installation.html#can-i-host-lychee-with-a-subpath-with-nginx-like-httpsexampledevlychee) suggests the following nginx configuration to serve Lychee from a subdirectory:

```nginx
location ^~ /lychee {
    alias /var/www/lychee/public;
    index index.php;
    try_files $uri $uri/ @lychee;
    location ~ \.php$ {
        if (!-e $request_filename) {
            rewrite ^/lychee/?(.*)$ /lychee/index.php?/$1 last;
            break;
        }
        fastcgi_pass unix:/var/run/php/php7.4-fpm.sock;
        fastcgi_index index.php;
        include fastcgi_params;
        fastcgi_param SCRIPT_FILENAME $request_filename;
    }
}
location @lychee {
    rewrite /lychee/(.*)$ /lychee/index.php?/$1 last;
}
```

I'm new to nginx, and I'm not sure I fully understand how this works.

- Why the two separate `rewrite` rules, one in `location @lychee` invoked by `try_files` and one in `if (!-e $request_filename)`? I tried with the latter commented out and it still seemed to work.

- Why `location ~ \.php$`? AFAICS Lychee has only one PHP file, `index.php`.

- What is `fastcgi_index index.php` for? According to the [documentation](https://nginx.org/en/docs/http/ngx_http_fastcgi_module.html#fastcgi_index), it does something with `$fastcgi_script_name`. But I used `log_format` to view the value of `$fastcgi_script_name` and it looked correct (`/lychee/index.php`). Removing `fastcgi_index` didn't change anything.

- Why the `location ^~ /lychee` instead of `location /lychee`?

This seems to be working for me:

```nginx
    location /lychee {
      alias /var/www/selim/lychee/public;
      index index.php;
      try_files $uri $uri/ @lychee;
      location /lychee/index.php {
        include fastcgi-include.conf;
        fastcgi_pass unix:/var/run/php/php8.2-fpm.sock;
        fastcgi_param SCRIPT_FILENAME $request_filename;
      }
    }
    location @lychee {
      rewrite /lychee/(.*)$ /lychee/index.php?/$1 last;
    }
```

**Comment by ildyria** [comment]
Hi @jeanas I am sorry, but I am not familiar with that approach. Maybe @d7415  can help you on that one.
